python/GafferUI/CompoundEditor.py

```diff
@@ -998,3 +1040,126 @@ def __getSharedDragWidget( self ) :
 
 	# /nasty-fudge
 
+
+## Returns the preferred bound for a widget's window in NDC space
+# (relative to the available screen area). We use the relative area to
+# help with portability between platforms/window managers.
+# '-1' is used to indicate windows that are on the primary screen.
+# If a window is full screen, we still include the bound as otherwise
+# the resultant placement when the window is un-fullscreen'd can be a bit small.
+#
+# Note: We use bottom-left NDC coordinates to be consistent with Gaffer
+def _preferredBound( gafferWindow ) :
+
+	qWidget = gafferWindow._qtWidget()
+
+	window = qWidget.windowHandle()
+	if not window :
+		return {}
+
+	widgetScreen = window.screen()
+	widgetScreenNumber = QtWidgets.QApplication.desktop().screenNumber( qWidget )
+
+	if widgetScreen == QtWidgets.QApplication.primaryScreen():
+		widgetScreenNumber = -1
+
+	screenGeom = widgetScreen.availableGeometry()
+	screenW = float( screenGeom.width() )
+	screenH = float( screenGeom.height() )
+
+	# Ideally we'd use frameGeometry for portability, but have seen a variety
+	# of issues where the window has been made visible, but doesn't yet
+	# have a frame, as we rely on asking for the margins (as you can't
+	# setFrameGeometry) we end up with unreliable window placement. As such
+	# we compromise and store the widgets geometry instead.
+	windowGeom = qWidget.normalGeometry()
+
+	x = ( windowGeom.x() - screenGeom.x() ) / screenW
+	y = 1.0 - ( ( windowGeom.y() - screenGeom.y() ) / screenH )
+	w = windowGeom.width() / screenW
+	h = windowGeom.height() / screenH
+
+	return {
```

Comment:
Is bound the right name now that we're storing all these other properties? Not that I can think of a better name...state perhaps?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/3167/commits/4361ca224e657be1db373d6d0fe91d18a252c235